### PR TITLE
[tf_micro] Allow toolchain for generated projects to be set in makefile

### PR DESCRIPTION
**Pull request by cdknorow**
* toolchain_prefix and toolchain_root are now filled in the makefile template when supplied
* adding a library compile option to the makefile template

### CODE REVIEW

**Comment by cdknorow** [comment]
@googlebot I consent.

**Comment by cdknorow** [comment]
I'm working on being able to generate projects that compile after being generated. This allows us to set the compiler paths from the makefile.inc of the targets, so that when they are generated the makefile includes the correct compiler.

File: tensorflow/lite/micro/tools/make/Makefile
Comment (wangtz) [inline tensorflow/lite/micro/tools/make/Makefile]: is -fPIC mandatory?

File: tensorflow/lite/micro/tools/make/templates/Makefile.tpl
Comment (wangtz) [inline tensorflow/lite/micro/tools/make/templates/Makefile.tpl]: should -r be part of ARFLASGS?

File: tensorflow/lite/micro/tools/make/templates/Makefile.tpl
Comment (wangtz) [inline tensorflow/lite/micro/tools/make/templates/Makefile.tpl]: $( ) ?

File: tensorflow/lite/micro/tools/make/templates/Makefile.tpl
Comment (wangtz) [inline tensorflow/lite/micro/tools/make/templates/Makefile.tpl]: is this a lib fro the framework or for the runnable binary?

If it's just a lib, it doesn't need to be part of generated project, unless we have a rule for generated lib project (e.g. without example code).

**Review by wangtz — COMMENTED** [review]
Thanks for the PR. 
I like the idea of making generated project buildable in the generated project folder. I think this PR needs some work on polishing. I'll also add @petewarden since he's most familiar with the makefiles

File: tensorflow/lite/micro/tools/make/Makefile
Comment (cdknorow) [reply tensorflow/lite/micro/tools/make/Makefile]: I'm unable to compile tensorflow-litemicro.a file into a shared object library without -fPIC, however, If that is not something that needs to be supported I can take it out.

File: tensorflow/lite/micro/tools/make/templates/Makefile.tpl
Comment (cdknorow) [reply tensorflow/lite/micro/tools/make/templates/Makefile.tpl]: hmm, this is a good point. Actually, what I want is to generate the project without example code. I was using this in my own example. I'll update this to build the library by ignoring the example directory. This way the library can be built and linked against for any of the projects.

File: tensorflow/lite/micro/tools/make/helper_functions.inc
Comment (cdknorow) [reply tensorflow/lite/micro/tools/make/helper_functions.inc]: got it, thanks!

File: tensorflow/lite/micro/tools/make/templates/Makefile.tpl
Comment (cdknorow) [reply tensorflow/lite/micro/tools/make/templates/Makefile.tpl]: thanks, I'll update that

File: tensorflow/lite/micro/tools/make/Makefile
Comment (wangtz) [inline tensorflow/lite/micro/tools/make/Makefile]: nit: new line at the end.

File: tensorflow/lite/micro/tools/make/Makefile
Comment (wangtz) [reply tensorflow/lite/micro/tools/make/Makefile]: I'm a bit confused, I thought .so is a shared library while .a (applies to us) is statically linked in.

Regardless from documentation, it looks that -fPIC is desirable. I'm OK with leaving it as it is.

**Review by wangtz — COMMENTED** [review]
Look good to me. Please wait for Pete's approval though. He has way more knowledge than me on makefiles. :)

File: tensorflow/lite/micro/tools/make/Makefile
Comment (petewarden) [reply tensorflow/lite/micro/tools/make/Makefile]: I think we should remove -fPIC here, and add it back in to particular platforms that need it, since in my experience it hasn't been very portable.

File: tensorflow/lite/micro/tools/make/Makefile
Comment (petewarden) [inline tensorflow/lite/micro/tools/make/Makefile]: Is this just in here for debugging purposes? We should remove it here I think.

**Review by petewarden — CHANGES_REQUESTED** [review]
Two small changes requested, otherwise looks good.

**Comment by cdknorow** [comment]
@petewarden commit updated.

**Review by petewarden — APPROVED** [review]

**Pull request merged**
